Data/Sequence.hs

```diff
@@ -13,34 +16,100 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- General purpose finite sequences.
--- Apart from being finite and having strict operations, sequences
--- also differ from lists in supporting a wider variety of operations
--- efficiently.
+-- = Finite sequences
+--
+-- The @'Seq' a@ type represents a finite sequence of values of
+-- type @a@.
+--
+-- Sequences generally behave very much like lists.
+--
+-- * The class instances for sequences are all based very closely on those for
+-- lists.
+--
+-- * Many functions in this module have the same names as functions in
+-- the "Prelude" or in "Data.List". In almost all cases, these functions
+-- behave analogously. For example, 'filter' filters a list in exactly the
+-- same way that 'Prelude.filter' filters a sequence. The only major exception
+-- is the 'lookup' function, which is based on the function by that name in
+--  "Data.Map" rather than the one from "Data.List".
+--
+-- There are two major differences between sequences and lists:
+--
+-- * Sequences support a wider variety of efficient operations than
+-- do lists. Notably, they offer
+--
+--     * Constant-time access to both the front and the rear, with
+--     '<|', '|>', 'viewl', 'viewr', and the bidirectional pattern
+--     synonyms ':<|' and ':|>'.
```

Comment:
Maybe mention that :<| and :|> are the preferred way?

Reply:
OK.